x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/persistence/ElasticsearchMappings.java

```diff
@@ -964,4 +985,95 @@ public static XContentBuilder auditMessageMapping() throws IOException {
                     .endObject()
                 .endObject();
     }
+
+    static String[] mappingRequiresUpdate(ClusterState state, String[] concreteIndices, Version minVersion,
+                                          Logger logger) throws IOException {
```

Comment:
I wonder if we need to pass down an external logger or just use the logger of `ElasticsearchMappings.class` instead.

Reply:
Sure, I changed that in the second commit.